storagemarket/impl/provider.go

```diff
@@ -392,6 +388,17 @@ func (p *providerDealEnvironment) Ask() storagemarket.StorageAsk {
 	return *sask.Ask
 }
 
+func (p *providerDealEnvironment) OnDataTransferEvent(ctx context.Context, eventCode datatransfer.EventCode, cb providerstates.OnDataTransferEventCallback) datatransfer.Unsubscribe {
```

Comment:
same problem as client. Again, I think this is better done via a single subscriber and reading the voucher to know where to route the event

Reply:
it generally should work with the existing DataTransferSubscriber with only a couple modifications